src/binding/http/serializer.rs

```diff
@@ -0,0 +1,77 @@
+use std::{cell::RefCell, rc::Rc};
+
+use crate::binding::{
+    http::{header_prefix, SPEC_VERSION_HEADER},
+    CLOUDEVENTS_JSON_HEADER,
+};
+use crate::event::SpecVersion;
+use crate::message::{BinarySerializer, MessageAttributeValue, Result, StructuredSerializer};
+
+macro_rules! str_to_header_value {
+    ($header_value:expr) => {
+        http::header::HeaderValue::from_str(&$header_value.to_string()).map_err(|e| {
+            crate::message::Error::Other {
+                source: Box::new(e),
+            }
+        })
+    };
+}
+
+pub trait Builder<R> {
+    fn header(&mut self, key: &str, value: http::header::HeaderValue);
```

Comment:
How would that make the calling code easier?

Reply:
It would remove the need for Rc/Cell/RefCell stuff and borrowing the builder, but I'm not sure if the final solution would be indeed simpler.

Reply:
Ah! Well, since I don't fully grok why we need the Rc/Cell/RefCell stuff, it's no wonder I can't see how returning the Option would remove that need. :)

I'm open to anything if you want to give it a shot.